Code/GraphMol/FileParsers/MolSGroupWriting.cpp

```diff
@@ -636,6 +636,37 @@ std::string FormatV3000AttachPointBlock(
   return ret.str();
 }
 
+namespace {
+void addBlockToSGroupString(std::string block, std::string &currentLine,
+                            std::ostringstream &os) {
+  if (currentLine.length() + block.length() < 78) {
+    currentLine += block;
+  } else {
+    os << currentLine << " -\n";
+    if (block.length() < 73) {
+      os << "M  V30" << block << " -\n";
+      currentLine = "M  V30";
+    } else {
+      unsigned int length = block.size();
+      unsigned int start = 0;
+      os << "M  V30" << block.substr(start, 72) << "-\n";
+      start += 72;
```

Comment:
This is redundant with the loop, isn't it?

Reply:
I don't think so. How do you mean?

Reply:
Sorry, I meant that these two lines are not required, since they do the same as the loop would do if they were not present.

Reply:
got it. I fixed that now.